functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy

```diff
@@ -518,4 +525,49 @@ abstract class BaseContainer extends Specification implements ClientProvider, Wa
         response.successful ? MAPPER.readValue(response.body().string(), valueType) :  unsuccessfulResponseHandler(response)
     }
 
+
+    void executeDockerActionOnRundeckContainer(String action) {
+        def process = "docker ${action} ${RUNDECK_CONTAINER_ID}".execute()
+        def stdout = new StringWriter()
+        def stderr = new StringWriter()
+        process.consumeProcessOutput(stdout, stderr)
+
+        def code = process.waitFor()
+
+        if (code != 0) {
+            throw new RuntimeException("Failed to ${action} container ${RUNDECK_CONTAINER_ID} with code ${code} - ${stderr.toString()}")
+        }
+    }
+
+
+
+    void waitForRundeckAppToBeResponsive(){
+
+        def checkIsRundeckApiResponding = {
+            try {
+                def response = client.doGet("/system/info")
+                return (response != null && response.code() == 200)
+            } catch (Exception e) {
+                LoggerFactory.getLogger(BaseContainer.class).error(e.getMessage())
```

Comment:
You do expect to see a few of these log lines in a "normal" flow, right?
In that case, the `error` level is a bit too harsh. 

```suggestion
                LoggerFactory.getLogger(BaseContainer.class).info(e.getMessage())
```